src/components/TransactionHistoryRows/TransactionGenericRow.tsx

```diff
@@ -176,71 +208,32 @@ export const TransactionGenericRow = ({
               </Box>
             }
           >
-            {assets.map((asset, index) => (
-              <Stack
-                alignItems='center'
-                key={index}
-                flex={1}
-                mt={{ base: 2, md: 0, xl: compactMode ? 2 : 0 }}
-                direction={index === 0 ? 'row' : 'row-reverse'}
-                textAlign={index === 0 ? 'left' : 'right'}
-              >
-                <AssetIcon
-                  assetId={asset.assetId}
-                  boxSize={{ base: '24px', lg: compactMode ? '24px' : '40px' }}
-                />
-                <Box flex={1}>
-                  <Amount.Crypto
-                    color='inherit'
-                    fontWeight='medium'
-                    prefix={index === 0 && isFirstAssetOutgoing ? '-' : ''}
-                    value={fromBaseUnit(asset.amount ?? '0', asset.precision)}
-                    symbol={asset.symbol}
-                    maximumFractionDigits={4}
-                  />
-                  {asset.currentPrice && (
-                    <Amount.Fiat
-                      color='gray.500'
-                      fontSize='sm'
-                      lineHeight='1'
-                      prefix={index === 0 && isFirstAssetOutgoing ? '-' : ''}
-                      value={bnOrZero(fromBaseUnit(asset.amount ?? '0', asset.precision))
-                        .times(asset.currentPrice)
-                        .toString()}
-                    />
-                  )}
-                </Box>
-              </Stack>
-            ))}
+            {DisplayTransfers}
           </Stack>
         </Flex>
         {isLargerThanLg && (
           <Flex alignItems='flex-start' flex={1} flexDir='column' textAlign='right'>
-            {fee && bnOrZero(fee?.amount).gt(0) ? (
+            {fee && bnOrZero(fee.value).gt(0) && (
               <Flex alignItems='flex-end' width='full'>
                 <Box flex={1}>
                   <Amount.Crypto
                     color='inherit'
                     fontWeight='bold'
-                    value={fromBaseUnit(fee.amount, fee.precision)}
-                    symbol={fee.symbol}
+                    value={fromBaseUnit(fee.value, fee.asset.precision)}
```

Comment:
Also nitpick and not introduced by this PR, but function calls should be passed with their return expression memoized as opposed to being directly passed down

If we're eventually going to use https://github.com/steadicat/eslint-plugin-react-memo for strict memoization, it's going to throw errors